ui/pages/confirmations/components/confirm/quote-swap-simulation-details/quote-swap-simulation-details.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useMemo } from 'react';
+import { BigNumber } from 'bignumber.js';
+import { Box, BoxFlexDirection } from '@metamask/design-system-react';
+import { Hex } from '@metamask/utils';
+import { QuoteResponse } from '@metamask/bridge-controller';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { toHex } from '@metamask/controller-utils';
+
+import { TokenStandAndDetails } from '../../../../../store/actions';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { useConfirmContext } from '../../../context/confirm';
+import { SimulationDetailsLayout } from '../../simulation-details/simulation-details';
+import { BalanceChangeRow } from '../../simulation-details/balance-change-row';
+import { TokenAssetIdentifier } from '../../simulation-details/types';
+
+export const QuoteSwapSimulationDetails = ({
+  fiatRates,
+  quote,
+  sourceTokenAmount,
+  tokenDetails,
+}: {
+  fiatRates?: Record<Hex, number | undefined>;
+  quote?: QuoteResponse;
+  sourceTokenAmount?: string;
+  tokenDetails?: Record<Hex, TokenStandAndDetails>;
+}) => {
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext<TransactionMeta>();
+  const { id: transactionId } = currentConfirmation;
+
+  const { srcAssetBalanceChange, destAssetBalanceChange } = useMemo(() => {
+    if (!quote || !tokenDetails || !fiatRates) {
+      return {};
+    }
+    const { srcAsset, destAsset, destTokenAmount } = quote.quote;
+    return {
+      srcAssetBalanceChange: {
```

Comment:
Minor, lots of logic and inline math, could put this in a local `getBalanceChanges(quote: Quote)` function with some distinct variables? Would also reduce the size of the `useMemo`.